databricks/koalas/frame.py

```diff
@@ -85,7 +86,19 @@ def _index_columns(self):
                 for field in self._metadata.index_fields]
 
     def _reduce_for_stat_function(self, sfun):
-        sdf = self._sdf.select([sfun(self._sdf[col]).alias(col) for col in self.columns])
+        from inspect import signature
```

Comment:
@rxin  this is very magical, and it is hard to debug what is happening here. There are 2 ways you could go here, I believe:
 - have a separate `_reduce_for_stat_function_with_type` , for which `sfun` is expected to take 2 arguments like you are doing with `count` below
 - use `k.sSeries` (which can resolve their types).

I think that the first option is the easiest, given that this is a low-level wrapper for spark functions.

Reply:
I'm tempted to just keep it as in, and add some documentation.

The reason is I wouldn't create a whole new function because only one function depends on it...